src/api/middleware/files.js

```diff
@@ -0,0 +1,110 @@
+const HTTPStatus = require("http-status-codes");
+const fs = require("fs");
+const path = require("path");
+const util = require("util");
+const { MulterError } = require("multer");
+const multerConfig = require("../../config/multer");
+const { ErrorTypes } = require("./errorHandler");
+const cloudinary = require("cloudinary").v2;
+const env = require("../../config/env");
+const ValidationReasons = require("./validators/validationReasons");
+
+
+const save_folder = path.join(__dirname, "../../../public/");
+if (!fs.existsSync(save_folder)) fs.mkdirSync(save_folder);
+
+const parseError = (message) => message.toLowerCase().replace(" ", "-");
+
+const single = (field_name) => (req, res, next) => {
+    const upload = multerConfig.single(field_name);
+    upload(req, res, (error) => {
+        if (error || !req.file) {
+            let message = "required";
+            let param = field_name;
+            if (error) {
+                message = error instanceof MulterError ?
+                    parseError(error.message) : error.message;
+                param = error.field ? error.field : param;
+            }
+            return res
+                .status(HTTPStatus.UNPROCESSABLE_ENTITY)
+                .json({
+                    error_code: ErrorTypes.VALIDATION_ERROR,
+                    errors: [
+                        {
+                            location: "body",
+                            param,
+                            msg: message
+                        }]
+                });
+        } else {
+            return next();
+        }
+    });
+};
+
+const save = async (req, res, next) => {
```

Comment:
```suggestion
const saveLocal = async (req, res, next) => {
```
To make a clearer distinction between cloud and this one